File: src/main/resources/static/js/angularjs/service/kindService.js
app.service('kindService',function ($http) {

    this.getAllParentKinds = function (page,rows,pId) {
        return $http.get('../kind/findparent?page='+page+'&rows='+rows+'&pId='+pId);
    }

    this.addKind = function (entity,pId) {
        return $http.post('../kind/addkind?pId='+pId,entity);
    }

})

File: src/main/java/com/gitsck/service/KindService.java
package com.gitsck.service;

import com.gitsck.entity.PageResult;
import com.gitsck.entity.Result;
import com.gitsck.pojo.GKind;

public interface KindService {

    PageResult findAllParentKinds(int page,int rows,long pId);

    Result addKind(GKind gKind,long pId);

    Result findByParentId(long pId);

    Result findall();

}
